R/query_dataset.R

```diff
@@ -0,0 +1,57 @@
+#' Query a data set
+#'
+#' @description
+#' A short description...
+#'
+#' @inheritParams api_url
+#' @param method An API query method. Needs to be "GET"
+#'
+#' @return Data frame containing query results
+#' @export
+#'
+#' @examples
+#' query_dataset(example_id(), indicators = example_id("indicator"), page = 1, page_size=10)
+query_dataset <- function(
+    dataset_id,
+    indicators,
+    time_periods = NULL,
+    geographic_levels = NULL,
+    locations = NULL,
+    filter_items = NULL,
+    method = "GET",
+    dataset_version = NULL,
```

Comment:
Can be a separate issue (if so, then raise one and stick it in the user testing milestone), though we should add some handling for only specifying minor versions etc, e.g. if you always wanted to get the latest v1.x.x version of a data set, but never v2, how do you do that, what do you put in the dataset_version param to get it ?